app/scripts/metamask-controller.js

```diff
@@ -488,13 +484,26 @@ export default class MetamaskController extends EventEmitter {
     });
 
     this.selectedNetworkController = new SelectedNetworkController({
+      state: initState.SelectedNetworkController,
       messenger: this.controllerMessenger.getRestricted({
         name: 'SelectedNetworkController',
         allowedActions: ['NetworkController:getNetworkClientById'],
         allowedEvents: ['NetworkController:stateChange'],
       }),
     });
 
+    if (
+      !this.selectedNetworkController.getNetworkClientIdForDomain('metamask')
+    ) {
+      this.selectedNetworkController.setNetworkClientIdForMetamask(
+        this.networkController.state.selectedNetworkClientId,
+      );
+    }
+    const networkClient =
+      this.selectedNetworkController.getProviderAndBlockTracker('metamask');
+    this.provider = networkClient.provider;
+    this.blockTracker = networkClient.blockTracker;
```

Comment:
Does this mean we're now using the clientRegistry for the provider in all of our controllers? Is this desired at this stage?